src/lib/response-triggers/userlocation.ts

```diff
@@ -0,0 +1,323 @@
+import Trigger from '../../models/trigger';
```

Comment:
This file is getting huge, and doesn't keep the concerns separated between the service and the module, so to speak.

Reply:
would abstracting out whether the user is admin or not be worthwhile/helpful?